### PR TITLE
Add Ruff Linter Integration and Pre-Commit Hook

### PR DESCRIPTION
**Pull request by deepakdinesh1123**
# Overview
This PR implements the integration of the Ruff linter into our Python project and sets up a pre-commit hook to enforce code quality standards before commits are made.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+exclude: ^docs/
```
Comment (DevilsAutumn) [inline .pre-commit-config.yaml:1]: If I am not wrong, this prevents running pre-commit on any file inside docs dir?

Comment (deepakdinesh1123) [reply .pre-commit-config.yaml:1]: yeah, we can see if including docs in linters would be beneficial or not, this was done as an experiment we can check if mdx is also parsed by pre-comit.

Comment (DevilsAutumn) [reply .pre-commit-config.yaml:1]: okay, let me have a look.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.5.0
+  hooks:
+  - id: check-added-large-files
```
Comment (DevilsAutumn) [inline .pre-commit-config.yaml:7]: any specific reason for checking for large files? 🤔 I don't think thats required.
Also, adding `requirements-txt-fixer` and `check-yaml` might be helpful.

Comment (deepakdinesh1123) [reply .pre-commit-config.yaml:7]: This is a template that was taken from another project that I worked on to check the feasibility of including it in this project, you can make any modifications that you see fit.

Comment (DevilsAutumn) [reply .pre-commit-config.yaml:7]: okay 👍

**Pull request closed without merging**